src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperDescriptorProvider.cs

```diff
@@ -95,37 +94,12 @@ public TagHelperDescriptorProvider(IEnumerable<TagHelperDescriptor> descriptors)
 
         private IEnumerable<TagHelperDescriptor> ApplyRequiredAttributes(
             IEnumerable<TagHelperDescriptor> descriptors,
-            IEnumerable<string> attributeNames)
+            IEnumerable<KeyValuePair<string, string>> attributes)
         {
             return descriptors.Where(
-                descriptor =>
-                {
-                    foreach (var requiredAttribute in descriptor.RequiredAttributes)
-                    {
-                        // '*' at the end of a required attribute indicates: apply to attributes prefixed with the
-                        // required attribute value.
-                        if (requiredAttribute.EndsWith(
-                            RequiredAttributeWildcardSuffix,
-                            StringComparison.OrdinalIgnoreCase))
-                        {
-                            var prefix = requiredAttribute.Substring(0, requiredAttribute.Length - 1);
-
-                            if (!attributeNames.Any(
-                                attributeName =>
-                                    attributeName.StartsWith(prefix, StringComparison.OrdinalIgnoreCase) &&
-                                    !string.Equals(attributeName, prefix, StringComparison.OrdinalIgnoreCase)))
-                            {
-                                return false;
-                            }
-                        }
-                        else if (!attributeNames.Contains(requiredAttribute, StringComparer.OrdinalIgnoreCase))
-                        {
-                            return false;
-                        }
-                    }
-
-                    return true;
-                });
+                descriptor => descriptor.RequiredAttributes.All(
+                    requiredAttribute => attributes.Any(
+                        attribute => requiredAttribute.Matches(attribute.Key, attribute.Value))));
```

Comment:
`Matches` -> `IsMatch` (along the same lines as `TagHelperAttributeDescriptor.IsNameMatch()`)